src/components/atoms/UiTextarea/UiTextarea.vue

```diff
@@ -80,10 +102,45 @@ const inputHandler = (event: Event) => {
   emit('update:modelValue', el.value);
 };
 const resizeValue = computed(() => {
+  if (props.hasAutogrowing) {
+    return 'none';
+  }
   if (typeof props.resize !== 'boolean') {
     return props.resize;
   }
-  return props.resize ? 'both' : 'none';
+  return props.resize
+    ? 'both'
+    : 'none';
+});
+const textarea = ref<HTMLTextAreaElement | null>(null);
+const textareaSize:Size = reactive({
+  width: null,
+  height: null,
+});
+const setTextareaSize = (mutationList: MutationRecord[]) => {
+  const { style } = mutationList[0].target as HTMLElement;
+  const {
+    width, height,
+  } = style;
+  textareaSize.width = width;
+  textareaSize.height = height;
+};
+const rootStyle = computed(() => (props.hasAutogrowing ? {} : {
+  width: textareaSize.width,
+  height: textareaSize.height,
+}));
+const observer = new MutationObserver(setTextareaSize);
+onMounted(async () => {
+  if (props.hasAutogrowing) { return; }
+  await nextTick();
+  observer.observe(textarea.value!, {
```

Comment:
The better way to say Typescript with what type is working:
<img width="752" alt="image" src="https://github.com/infermedica/component-library/assets/37744371/93c11018-cb67-44ee-ba9a-417bd7ec9b65">

is `type assertion`

<img width="471" alt="image" src="https://github.com/infermedica/component-library/assets/37744371/236ff6b0-660d-4a53-bcfb-0d5a0511e8ca">

Reply:
:thank_you:

Reply:
I won't agree, assertion is rarely the better way to narrow the type 😅 The best way is to narrow the type with proper checks (e.g. type-guards) 😄

Reply:
so @AlanLes do you suggest this solution:
```
if (textarea.value) {
    observer.observe(textarea.value, {
      attributes: true,
      childList: false,
      subtree: false,
    });
  }
});
```
?

Reply:
I didn't check what properties need to be checked to narrow this type properly, but If it does satisfy the TS, then yes. :)